cmd/gctcli/websocket_management.go

```diff
@@ -0,0 +1,270 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/thrasher-corp/gocryptotrader/gctrpc"
+	"github.com/urfave/cli"
+)
+
+var websocketManagerCommand = cli.Command{
+	Name:      "websocket",
+	Usage:     "execute websocket management command",
+	ArgsUsage: "<command> <args>",
+	Subcommands: []cli.Command{
+		{
+			Name:  "getinfo",
+			Usage: "returns all exchange websocket information",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+			},
+			Action: getwebsocketInfo,
+		},
+		{
+			Name:  "disable",
+			Usage: "disables websocket connection for an exchange",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+			},
+			Action: enableDisableWebsocket,
+		},
+		{
+			Name:  "enable",
+			Usage: "enables websocket connection for an exchange",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+				cli.BoolTFlag{
+					Name:   "enable",
+					Hidden: true,
+				},
+			},
+			Action: enableDisableWebsocket,
+		},
+		{
+			Name:  "getsubs",
+			Usage: "returns current subscriptions for an exchange",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+			},
+			Action: getSubscriptions,
+		},
+		{
+			Name:  "setproxy",
+			Usage: "sets exchange websocket proxy, flushes and reroutes connection",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+				cli.StringFlag{
+					Name:  "proxy",
+					Usage: "proxy address to change to",
+				},
+			},
+			Action: setProxy,
+		},
+		{
+			Name:  "seturl",
+			Usage: "sets exchange websocket connection, flushes and reconnects",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "the exchange to act on",
+				},
+				cli.StringFlag{
+					Name:  "url",
+					Usage: "url string to change to, an empty string will set it back to default",
+				},
+			},
+			Action: setURL,
+		},
+	},
+}
+
+func getwebsocketInfo(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowSubcommandHelp(c)
+	}
+
+	var exchange string
+	if c.IsSet("exchange") {
+		exchange = c.String("exchange")
+	} else {
+		exchange = c.Args().First()
+	}
+
+	if !validExchange(exchange) {
+		return fmt.Errorf("[%s] is not a valid exchange", exchange)
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+	result, err := client.WebsocketGetInfo(context.Background(),
+		&gctrpc.WebsocketGetInfoRequest{Exchange: exchange})
+	if err != nil {
+		return err
+	}
+	jsonOutput(result)
+	return nil
+}
+
+func enableDisableWebsocket(c *cli.Context) error {
```

Comment:
When updating the enabled/disabled status, it does not affect at the config level. So when enabling, the `syncer.go` does not receive websocket updates when GCT starts with the websocket disabled. Secondly, it doesn't update the config when closing the application